### PR TITLE
feat: [CI-6093]: Added POJO for Script and Plugin Step for Yaml Simplification

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Added POJO for Script and Plugin Step for Yaml Simplification
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40397)

### CODE REVIEW

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+@ApiModel(subTypes = {ScriptStepInfo.class, PluginStepInfoV1.class})
+@OwnedBy(CI)
+public abstract class CIStepInfoV1 implements CIStepInfo {
+  @VariableExpression(skipVariableExpression = true) public static final int DEFAULT_RETRY = 1;
+  @JsonProperty(YamlNode.UUID_FIELD_NAME)
+  @Getter(onMethod_ = { @ApiModelProperty(hidden = true) })
+  @ApiModelProperty(hidden = true)
+  protected String uuid;
+
+  @Getter(onMethod_ = { @ApiModelProperty(hidden = true) })
```
Comment (harshjain12) [inline 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java:40]: Not required all 3

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  protected ParameterField<List<Volume>> volumes;
+  public ParameterField<List<Volume>> getVolumes() {
+    if (this.volumes.getValue() == null) {
+      this.volumes.setValue(Collections.emptyList());
+    }
+    return this.volumes;
+  }
+
+  @Override
+  public int getRetry() {
```
Comment (harshjain12) [inline 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java:62]: not required

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/PluginStepInfoV1.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  @YamlSchemaTypes({runtime})
+  @ApiModelProperty(dataType = BOOLEAN_CLASSPATH)
+  private ParameterField<Boolean> privileged;
+  @YamlSchemaTypes({string}) @ApiModelProperty(dataType = INTEGER_CLASSPATH) private ParameterField<Integer> user;
+  @YamlSchemaTypes({runtime})
+  @ApiModelProperty(dataType = "io.harness.beans.yaml.extended.beans.PullPolicy")
+  private ParameterField<PullPolicy> pull;
+
+  @Builder
+  @ConstructorProperties({"uuid", "identifier", "name", "retry", "with", "image", "uses", "resources", "outputs", "env",
```
Comment (harshjain12) [inline 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/PluginStepInfoV1.java:86]: Check if we can remove it

Comment (harshjain12) [reply 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/PluginStepInfoV1.java:86]: SuperBuilder

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/yaml/extended/platform/V1/PlatformV1.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import io.harness.yaml.YamlSchemaTypes;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
```
Comment (harshjain12) [inline 332-ci-manager/contracts/src/main/java/io/harness/cimanager/yaml/extended/platform/V1/PlatformV1.java:25]: make it @value in all pojos

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/V1/CIPlanCreatorUtils.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        ParameterField<String> repoName = optionalRepository.isEmpty()
+            ? ParameterField.createValueField(gitSyncBranchContext.getGitBranchInfo().getRepoName())
+            : optionalRepository.get().getName();
+        ParameterField<String> connector = optionalRepository.isEmpty()
+            ? ParameterField.createValueField(ctx.getMetadata().getMetadata().getPipelineConnectorRef())
+            : optionalRepository.get().getConnector();
+        codeBaseBuilder = codeBaseBuilder.repoName(repoName).connectorRef(connector);
+        break;
+      case INLINE:
+        if (optionalRepository.isEmpty()) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/V1/CIPlanCreatorUtils.java:96]: Put it in different method

Comment (raghavharness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/V1/CIPlanCreatorUtils.java]: done

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java (hunk 2 above)
Comment (raghavharness) [reply 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java]: done

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java (hunk 1 above)
Comment (raghavharness) [reply 332-ci-manager/contracts/src/main/java/io/harness/cimanager/stepinfo/V1/CIStepInfoV1.java]: done

**Review by williamwissemann — APPROVED** [review]
> Reviewed 29 of 42 files at r1, 14 of 14 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40397)* status: Waiting on team(s): sto.

**Pull request merged**